machine.go

```diff
@@ -496,3 +529,15 @@ func convertToStringSlice(field interface{}) []string {
 	}
 	return result
 }
+
+func convertToStringMap(field interface{}) map[string]string {
+	if field == nil {
+		return nil
+	}
+	fieldMap := field.(map[string]interface{})
```

Comment:
Yes, it's always guarded by schema coercion - I'll put in a comment to that effect.